framework/src/outputs/OutputWarehouse.C

```diff
@@ -90,7 +90,7 @@ OutputWarehouse::subdomainSetup()
 }
 
 void
-OutputWarehouse::addOutput(std::shared_ptr<Output> & output)
+OutputWarehouse::addOutput(std::shared_ptr<Output> const output)
```

Comment:
Why the `const` point qualifier not in the header?

Reply:
If at some point a developer wanted to intentionally modify this argument, the could remove the `const` in the implementation (.C) file without having to modify the header, causing recompilation of all the translation units that depend on it. As it does not matter to the caller whether this is marked as `const` or not (since it is pass-by-value) it really does not matter whether, from an API standpoint, you mark it as such. However, marking it `const` in the implementation helps to make sure you don't accidentally do something stupid in the implementation, like change `output` to point to some other object

Reply:
> As it does not matter to the caller whether this is marked as const or not (since it is pass-by-value) it really does not matter whether

👍 
Thanks for the details.